### PR TITLE
Feature/metrics

### PR DESCRIPTION
**Pull request by m-mcgowan**
Adds analytics/metrics support to commands supplied via particle-commands by providing the analytics details required in the command context. 

Depends upon https://github.com/spark/particle-commands/pull/15

### CODE REVIEW

**Comment by kennethlimcp** [comment]
Is this an `opt-in`, `opt-out` or compulsory feature?

**Comment by m-mcgowan** [comment]
It's presently undecided, but most likely opt-out

File: src/app/nested-yargs.js [hunk 1]
```diff
@@ -641,8 +642,20 @@ function showHelp() {
 	Yargs.showHelp();
 }
 
+/**
+ * Centralizes command execution.
+ */
+class Executor {
```
Comment (monkbroc) [inline src/app/nested-yargs.js]: Executor class!

**Comment by monkbroc** [comment]
Let's close this and revisit later.

**Pull request closed without merging**